tests/integration/utils.py

```diff
@@ -0,0 +1,94 @@
+import configparser
+import os
+from spccore.connection import *
+
+
+DEFAULT_CONFIG_PATH = os.path.join(os.path.expanduser('~'), '.synapseConfig')
+DEFAULT_CONFIG_ENDPOINT_SECTION = "endpoints"
+DEFAULT_CONFIG_AUTH_SECTION = "authentication"
+DEFAULT_CONFIG_USERNAME_OPT = "username"
+DEFAULT_CONFIG_PASSWORD_OPT = "password"
+DEFAULT_CONFIG_REPO_ENDPOINT_OPT = "repoEndpoint"
+DEFAULT_CONFIG_AUTH_ENDPOINT_OPT = "authEndpoint"
+DEFAULT_CONFIG_FILE_ENDPOINT_OPT = "fileHandleEndpoint"
+
+
+def get_anonymous_connection() -> SynapseConnection:
+    """
+    Retrieve the dev stack endpoint
+
+    :return: the dev stack endpoint
+    """
+    repo, auth, file = get_test_endpoints()
+    return get_connection(repo_endpoint=repo, auth_endpoint=auth, file_endpoint=file)
+
+
+def get_test_user_connection() -> SynapseConnection:
+    """
+    Get a connection using the credentials found in the config file
+
+    :return: the test user's connection to Synapse
+    """
+    repo, auth, file = get_test_endpoints()
+    username, password = get_test_credentials()
+    api_key = _get_api_key(repo, auth, file, username, password)
+    return get_connection(repo_endpoint=repo,
+                          auth_endpoint=auth,
+                          file_endpoint=file,
+                          username=username,
+                          api_key=api_key)
+
+
+def get_test_endpoints() -> (str, str, str):
+    """
+    Read ~/.synapseConfig and retrieve endpoint, test username and password
+
+    :return: repo, auth, and file endpoints
+    """
+    config = configparser.ConfigParser()
+    config.read(DEFAULT_CONFIG_PATH)
```

Comment:
see if this keeps the config file's data in memory or if it reads from disk each time. we may want to cache the contents of the config